test/binding-redirects.Tests/BindingRedirectTests.cs

```diff
@@ -0,0 +1,287 @@
+                value.Value.Should().Be(appSetting.Value, $"For AppSetting '{appSetting.Key}' - Expected Value '{appSetting.Value}', Actual '{ value.Value}'");
+            }
+        }
+
+        [Fact]
+        public void Build_Generates_Redirects_For_App_Without_Config()
+        {
+            var redirects = GetRedirects(_appWithoutConfigBuildOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppNoConfig);
+
+            var commandResult = new TestCommand(_appWithoutConfigBuildOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        [Fact]
+        public void Publish_Generates_Redirects_For_App_Without_Config()
+        {
+            var redirects = GetRedirects(_appWithoutConfigPublishOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppNoConfig);
+
+            var commandResult = new TestCommand(_appWithoutConfigPublishOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        [Fact]
+        public void Build_Generates_Redirects_For_Executable_Dependency()
+        {
+            var redirects = GetRedirects(_executableDependencyBuildOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppNoConfig);
+
+            var commandResult = new TestCommand(_executableDependencyBuildOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/cli/issues/2632")]
+        public void Publish_Generates_Redirects_For_Executable_Dependency()
+        {
+            var redirects = GetRedirects(_executableDependencyPublishOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppNoConfig);
+
+            var commandResult = new TestCommand(_executableDependencyPublishOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        [Fact]
+        public void Build_Generates_Redirects_For_App_With_Config()
+        {
+            var redirects = GetRedirects(_appWithConfigBuildOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppWithConfig);
+            VerifyAppSettings(_appWithConfigBuildOutput);
+
+            var commandResult = new TestCommand(_appWithConfigBuildOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        [Fact]
+        public void Publish_Generates_Redirects_For_App_With_Config()
+        {
+            var redirects = GetRedirects(_appWithConfigPublishOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppWithConfig);
+            VerifyAppSettings(_appWithConfigPublishOutput);
+
+            var commandResult = new TestCommand(_appWithConfigPublishOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+    }
```

Comment:
We need two more tests. We should invoke the app via the Command Resolver, ideally through the dotnet-dependency-tool-invoker, and show that the binding redirects are working correctly in both the config and no-config case.